spec/requests/purchases_spec.rb

```diff
@@ -0,0 +1,114 @@
+require 'rails_helper'
+
+RSpec.describe 'Purchases', type: :request do
+  let(:group) { Group.create(user: @user, name: 'Food', icon: 'missing_avatar.png') }
+  let(:purchase) { Purchase.create(name: 'Apples', amount: 5, author: @user) }
+  let(:group_purchase) { GroupPurchase.create(group: group, purchase: purchase) }
+  let(:valid_attributes) { { 'name' => 'Bananas', 'amount' => 5, 'author' => @user, 'group_ids' => [group.id] } }
+
+  before :each do
+    @user = User.create(name: 'Tom', email: 'tom@example.com', password: 'topsecret')
+    @user.confirm
```

Comment:
- [ ] Most of your tests are failing. Kindly fix them.

  ![image](https://github.com/katarighe/budget-app/assets/54848363/9fc067b7-23dd-4db3-80a2-639588658cc0)